api/kyverno/v1alpha1/cleanup_policy_types.go

```diff
@@ -109,3 +121,198 @@ type CleanupPolicySpec struct {
 type CleanupPolicyStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
 }
+
+// Validate implements programmatic validation
+func (p *CleanupPolicy) Validate(clusterResources sets.String) (errs field.ErrorList) {
+	errs = append(errs, kyvernov1.ValidateAutogenAnnotation(field.NewPath("metadata").Child("annotations"), p.GetAnnotations())...)
```

Comment:
Is there autogen support ?

Reply:
I don't think we should support autogen for the cleanup policy, the pod and pod-controller may not have the same age.

Reply:
Agree, we discussed it on slack with @NikhilSharmaWe 
I think he is going to remove it

Reply:
Removed the check for auto gen in cleanupPolicy.